apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.js

```diff
@@ -104,43 +103,33 @@ router.post(
     } else if (req.body.__action === 'add_group') {
       const assessment_id = res.locals.assessment.id;
       const group_name = req.body.group_name;
-      if (!group_name || String(group_name).length < 1) {
-        flash('error', 'Group name cannot be empty.');
-        res.redirect(req.originalUrl);
-        return;
-      }
 
       const uids = req.body.uids;
       const uidlist = uids.split(/[ ,]+/).filter((uid) => !!uid);
-      if (uidlist.length === 0) {
-        flash('error', 'Group must be created with at least one user.');
-        res.redirect(req.originalUrl);
-        return;
+      const userIdList = [];
+      const notExist = [];
+      for (const uid of uidlist) {
+        const user = await selectUserByUid(uid);
```

Comment:
This would allow someone to enumerate emails that exist within PrairieLearn. There may already be other ways to do this, so I don't know how critical it is to avoid this (cc @mwest1066 what do you think?). I know `createGroup` will transitively check enrollment status, but you should consider limiting this check to users that are enrolled in the course instance.

Now that I'm paying attention to this, `uploadInstanceGroups` suffers from the same problem.

Reply:
I agree that we mustn't leak UIDs, even to instructors at the same institution. This is a hard requirement for some parts of FERPA (the so-called "FERPA-suppressed" students). This is the reason for PTs "invite" system for enrollments/course staff/etc -- it doesn't leak anything about the validity of a UID.

Reply:
Would it make sense to have the error thrown if the user does not exist be exactly the same as if the user is not enrolled?

Reply:
Yep, just saying "there is not such student in the class" is all we need in both cases, I think.